### DISCUSSION TITLE
TwoFactor or Multifactor Authentication

### DISCUSSION THREAD

**Discussion by canuck68** [epoupon/fileshelter#47] [General]
Hi, love the application, but we would need to do two-factor authentication with ldap/duo security to use it in our environment.  Any idea how you could add authentication before getting to the fileshelter service ?

Thanks,
/casey

**Comment by epoupon** [comment]
Hello!
Have you tried to check what it is possible to do with a reverse proxy?

**Reply by markusglaetzner** [reply]
> Hi, love the application, but we would need to do two-factor authentication with ldap/duo security to use it in our environment. Any idea how you could add authentication before getting to the fileshelter service ?
> 
> Thanks, /casey

Hi, you can use authelia for 2fa.

**Reply by markusglaetzner** [reply]
> Hello! Have you tried to check what it is possible to do with a reverse proxy?

It works great with a reverse proxy (nginx for example).

**Reply by viharm** [reply]
For _Authelia_, does it work with the standard `Remote-User` header?

**Comment by canuck68** [comment]
I tried but I had to move on because the we the requirements insisted we do
2FA.  Would've loved to have got it going and use it, but could not get the
2FA going.

/casey

**Comment by canuck68** [comment]
I did not go any further and my boss opted for another fileshare.  Too  ad
b/c I really liked your app.
